pkgs/by-name/cr/cryptographic-id-rs/package.nix

```
@@ -0,0 +1,53 @@
+{
+  lib,
+  fetchFromGitLab,
+  rustPlatform,
+  libclang,
+  linuxHeaders,
+  pkg-config,
+  tpm2-tss,
+  protobuf,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "cryptographic-id-rs";
+  version = "0.4.1";
+
+  src = fetchFromGitLab {
+    owner = "cryptographic_id";
+    repo = "cryptographic-id-rs";
+    tag = "v${finalAttrs.version}";
+    fetchSubmodules = true;
+    hash = "sha256-cKy3psFXV5LkUHC9UaF6QOee1kr/OXm1k2LxDCBypOA=";
+  };
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-xlP6u+c8R6c1aevod1CwBN+8Y2M/f2jvKzWUHrgqmVM=";
+
+  nativeBuildInputs = [
+    rustPlatform.bindgenHook
+    pkg-config
+    linuxHeaders
+    protobuf
+  ];
+
+  buildInputs = [
+    tpm2-tss
+  ];
+
+  # tests are broken without access to TPM / swTPM
+  doCheck = false;
+
+  env = {
+    LIBCLANG_PATH = "${lib.getLib libclang}/lib";
+  };
+
```

sorry, forgot to mention
```suggestion

  passthru.updateScript = nix-update-script { };

```

---

done, thanks